contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -363,12 +369,21 @@ public void emitTuples()
     if (maxTuplesPerWindow > 0) {
       count = Math.min(count, maxTuplesPerWindow - emitCount);
     }
+    if (count > 0) {
+      // Transmit at least one message in the window even if the message size is greater than max message size
+      if ((emitCount > 0) && ((maxMsgSizePerWindow - emitMsgSize) < consumer.peekMessage().msg.size())) {
+        return;
+      }
+    }
```

Comment:
There are two exit conditions from the loop, one in which i is incremented and one not, more readable to just go with a separate variable instead of uniformizing the value of i on exit.